packer/provisioner.go

```diff
@@ -145,17 +145,55 @@ func (p *PausedProvisioner) Prepare(raws ...interface{}) error {
 
 func (p *PausedProvisioner) Provision(ctx context.Context, ui packersdk.Ui, comm packersdk.Communicator, generatedData map[string]interface{}) error {
 
-	// Use a select to determine if we get cancelled during the wait
 	ui.Say(fmt.Sprintf("Pausing %s before the next provisioner...", p.PauseBefore))
-	select {
-	case <-time.After(p.PauseBefore):
-	case <-ctx.Done():
-		return ctx.Err()
+	minimumTimeForPauseUpdate := float64(10)
+	if p.PauseBefore.Seconds() < minimumTimeForPauseUpdate {
+		// Use a select to determine if we get cancelled during the wait
+		select {
+		case <-time.After(p.PauseBefore):
+		case <-ctx.Done():
+			return ctx.Err()
+		}
+	} else {
+		err := p.updatesWhilePausing(ctx, ui)
+		if err != nil {
+			return err
+		}
 	}
 
 	return p.Provisioner.Provision(ctx, ui, comm, generatedData)
 }
 
+func (p *PausedProvisioner) updatesWhilePausing(ctx context.Context, ui packersdk.Ui) error {
+	updateTime := 10
+	timeTicker := time.NewTicker(time.Duration(updateTime) * time.Second)
+	TotalTime := p.PauseBefore.Seconds()
+	tickerChannel := make(chan bool)
+	var err error
+	go func() {
+		for {
+			select {
+			case <-timeTicker.C:
+				TotalTime -= float64(updateTime)
+				ui.Say(fmt.Sprintf("%v seconds left until the next provisioner", TotalTime))
+			case <-ctx.Done():
+				err = ctx.Err()
+				return
+			case <-tickerChannel:
+				return
+			}
+		}
+	}()
+	time.Sleep(p.PauseBefore)
+	timeTicker.Stop()
+	if err != nil {
+		return err
+	}
+	tickerChannel <- true
+
+	return nil
```

Comment:
You can simplify this a bit and remove the `tickerChannel` channel:
```suggestion
	var err error
	go func() {
		for {
			select {
			case <-timeTicker.C:
				TotalTime -= float64(updateTime)
				ui.Say(fmt.Sprintf("%v seconds left until the next provisioner", TotalTime))
				if TotalTime <= float64(updateTime) {
					return
				}
			case <-ctx.Done():
				err = ctx.Err()
				return
			}
		}
	}()
	time.Sleep(p.PauseBefore)
	timeTicker.Stop()
	return err
```